beacon_node/client/src/builder.rs

```diff
@@ -354,15 +354,10 @@ where
                 let anchor_block = SignedBeaconBlock::from_ssz_bytes(&anchor_block_bytes, &spec)
                     .map_err(|e| format!("Unable to parse weak subj block SSZ: {:?}", e))?;
 
-                // `BlobSidecar` is no longer used from Fulu onwards (superseded by `DataColumnSidecar`),
-                // which will be fetched via rpc instead (unimplemented).
-                let is_before_fulu = !spec
-                    .fork_name_at_slot::<E>(anchor_block.slot())
-                    .fulu_enabled();
-                let anchor_blobs = if is_before_fulu && anchor_block.message().body().has_blobs() {
+                // Providing blobs is optional now and not providing them is recommended.
+                // Backfill can handle downloading the blobs or columns for the checkpoint block.
+                let anchor_blobs = if let Some(anchor_blobs_bytes) = anchor_blobs_bytes {
```

Comment:
Sorry for the verbose reviews above. Here are the key points concisely:

1. Missing test for None blobs path - existing test at `store_tests.rs:2874` still provides blobs
2. Consider logging when blobs omitted but block has them (aids debugging)
3. Check if these docs need updating: `book/src/advanced_checkpoint_sync.md`, `advanced_blobs.md`, CLI help for `--checkpoint-blobs`

Otherwise looks good - builds correctly on #8417.

Reply:
Test added here https://github.com/sigp/lighthouse/pull/8473/files

Reply:
Thanks!